rpc/src/client/compat.rs

```diff
@@ -0,0 +1,81 @@
+//! Support for dynamic compatibility with older protocol versions.
+
+use tendermint::Version;
+
+use crate::prelude::*;
+use crate::Error;
+
+/// Protocol compatibility mode for a Tendermint RPC client.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum CompatMode {
+    /// Use the latest version of the protocol (v0.37 in this release).
+    Latest,
```

Comment:
What's the reasoning behind using `Latest` here instead of `V0_37`? 

Having the datatype variant change meaning over releases sounds brittle to me, and it would be more explicit to have it denote the actual release that is supported.

If there are no good reasons against it, how about we rename this to `V0_37` and add a `CompatMode::latest()` method that returns the latest compat mode. Let's also change the `Default` instance to return `CompatMode::latest()`.

Reply:
Good idea. There may be a case for the `Latest` variant alongside both explicitly selective versions, but perhaps this will produce complications on matching. So let's do "latest" as an associated const or fn.

Reply:
Redone in https://github.com/informalsystems/tendermint-rs/pull/1193/commits/f3e161e02b4bc42a9c60a369a4b0b7ecd779da99